src/som/interpreter/actors/EventualMessage.java

```diff
@@ -233,13 +245,15 @@ public SPromise getPromise() {
   /** The callback message to be send after a promise is resolved. */
   public static final class PromiseCallbackMessage extends PromiseMessage {
     protected final SPromise promise;
+    protected final SPromise originalPromise;
```

Comment:
I don't know what `originalPromise` is suppose to mean.
How does it relate to the `promise`? These names need clarification, I think.